### ISSUE TITLE
AWS Rekognition focal point detection result causes SQL error for some images

### ISSUE THREAD

**Issue by JoshCoady** [craftcms/aws-s3#101]
### Description

I'm getting this error message on some images, not all. Likely not expecting a negative focal point coord due to the chopped face?

```
SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'focalPoint'

The SQL being executed was: UPDATE `craft_assets` SET `focalPoint`='0.7921;-0.0558', 
`dateUpdated`='2020-09-26 03:37:08' WHERE `id`=2662 
```

Example image that was being uploaded to produce the above: https://storage.googleapis.com/ltkcms.appspot.com/fs/yd/images/cover/woman-vaccine.base.webp?v=1583787691

### Additional info

- Craft version: 3.5.9 pro
- PHP version: 7.4.4
- Database driver & version: MySQL 5.5.5
- Plugins & versions: S3 1.2.11

**Comment by JoshCoady** [comment]
Dont know if this is the best method, but it fixes it.

`Volume::detectFocalPoint`:
```
                return [
                    number_format(max(0, $box['Left'] + ($box['Width'] / 2)), 4),
                    number_format(max(0, $box['Top'] + ($box['Height'] / 2)), 4),
                ];
```

**Comment by JoshCoady** [comment]
Workaround:
```
        Event::on(
            \craft\records\Asset::class,
            \craft\records\Asset::EVENT_BEFORE_UPDATE,
            function(\yii\base\ModelEvent $event) {
                /** @var \craft\records\Asset $image */
                $image = $event->sender;
                if($image->focalPoint) {
                    $fp = explode(';', $image->focalPoint);
                    $fp[0] = max(0, (float)$fp[0]);
                    $fp[1] = max(0, (float)$fp[1]);
                    $image->focalPoint = implode(';', $fp);
                }
            }
        );
```

**Comment by brandonkelly** [comment]
Guessing this could be fixed by changing this line:

https://github.com/craftcms/aws-s3/blob/d9934e0d34c3a1617a5e120d58b7062f8d302751/src/Plugin.php#L69

to:

```php
$assetRecord->focalPoint =  min(max($focalPoint[0], 0), 1) . ';' .  min(max($focalPoint[1], 0), 1);
```

**Comment by andris-sevcenko** [comment]
Yeah, ended up doing just that.

Resolved this in d844973d60f7bb163b8d88d4d72706886cb5e1dd
